airflow/dags/dags.py

```diff
@@ -114,6 +115,24 @@ def sql_enrich_duplicates(schema_tbl, key_columns, order_by_columns):
     """
 
 
+def email_failure(context):
+    slack_url = os.environ.get("CALITP_SLACK_URL")
+    if slack_url is None:
+        print("Skipping email to slack channel. No CALITP_SLACK_URL in environment")
+
+    ti = context["ti"]
+    message = f"""
+Task Failed: {ti.dag_id}.{ti.task_id}
+Execution Date: {ti.execution_date}
+Try {ti.try_number} of {ti.max_tries}
+
+<{ti.log_url}| Check Log >
+"""
+    requests.post(
+        os.environ["CALITP_SLACK_URL"], json={"text": message},
+    )
```

Comment:
Could this result in an error if it fails? Should we surround it with try except?

Reply:
@atvaccaro

Reply:
In the past, I have sometimes found issues with accessing the task instance information from the context. I was never really sure _why_; I would recommend an overall structure similar to the following.

```python
try:
    msg = "..."
    requests.post(...)
except:
    requests.post("failed to log to slack")
```

With the `except` being as dumbed-down as possible as a last-ditch attempt to log to Slack.